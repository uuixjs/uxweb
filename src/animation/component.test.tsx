import { AnimationEvent } from "react";
import { setupShallowTest } from "../../tests/helpers";
import {
  Animation,
  AnimationDelay,
  AnimationDuration,
  AnimationFillMode,
  AnimationProps,
  AnimationTiming,
  AnimationType,
} from "./component";

function getRequiredProps(): AnimationProps {
  return {
    type: AnimationType.Bounce,
  };
}

function getOptionalProps(): AnimationProps {
  return {
    delay: AnimationDelay.Medium,
    duration: AnimationDuration.Medium,
    enabled: true,
    fillMode: AnimationFillMode.Both,
    loop: true,
    timing: AnimationTiming.EaseInOut,
    type: AnimationType.Bounce,
    onAnimationStart: jest.fn(),
    onAnimationEnd: jest.fn(),
  };
}

const setupShallow = setupShallowTest(Animation, getRequiredProps);

describe("Animation", () => {
  it("renders animation component with only required props", () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders animation component with all props", () => {
    const { wrapper } = setupShallow(getOptionalProps());
    expect(wrapper).toMatchSnapshot();
  });

  it("sets the onAnimationStart prop on the correct element", () => {
    const onAnimationStart = jest.fn();
    const { wrapper } = setupShallow({ onAnimationStart });

    wrapper.props().onAnimationStart!({} as AnimationEvent<HTMLDivElement>);

    expect(onAnimationStart).toBeCalled();
  });

  it("sets the onAnimationEnd prop on the correct element", () => {
    const onAnimationEnd = jest.fn();
    const { wrapper } = setupShallow({ onAnimationEnd });

    wrapper.props().onAnimationEnd!({} as AnimationEvent<HTMLDivElement>);

    expect(onAnimationEnd).toBeCalled();
  });
});
