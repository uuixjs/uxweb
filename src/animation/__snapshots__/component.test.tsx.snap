// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation renders animation component with all props 1`] = `
<styled.div
  $delay="medium"
  $duration="medium"
  $enabled={true}
  $fillMode="both"
  $loop={true}
  $timing="ease-in-out"
  $type="bounce"
  className="tw-animation"
  data-a-target="tw-animation-target"
  onAnimationEnd={[MockFunction]}
  onAnimationStart={[MockFunction]}
/>
`;

exports[`Animation renders animation component with only required props 1`] = `
<styled.div
  $duration="short"
  $fillMode="both"
  $timing="ease"
  $type="bounce"
  className="tw-animation"
  data-a-target="tw-animation-target"
/>
`;
